### ISSUE TITLE
Please considering adding AgentSquare to the repository

### ISSUE THREAD

**Issue by THU-Yu-Shang** [WooooDyy/LLM-Agent-Paper-List#70]
Thanks for the collection of works about LLM agents, I'd like to kindly request that you consider adding our work **AgentSquare** to your repository.

AgentSquare is a novel LLM agent search framework that utilizes module evolution and recombination to automatically optimize agent designs across a modular design space consisting of Planning, Reasoning, Tool Use, and Memory modules. We believe that the proposed modular design space and AgentSquare search framework offer a platform for fully exploiting the potential of prior successful designs and consolidating the collective efforts of the research community of LLM agents. 

Here's some information about AgentSquare:
**AgentSquare: Automatic LLM Agent Search in Modular Design Space**
**Website (with demos):** https://tsinghua-fib-lab.github.io/AgentSquare_website/
**Paper:** https://arxiv.org/abs/2410.06153
**Code:** https://github.com/tsinghua-fib-lab/AgentSquare

**TLDR:** We propose a modular design space of LLM agents and an automatic agent search framework named AgentSquare, which can adaptively search for better-performing agent designs for given tasks.

**Key Features:**
**Unified Framework:** We have provided unified exemplar scripts on six popular agent tasks covering embodied, game, web and tool areas. You can easily run different agent designs with only a line of code!
**Extensibility:** We have provided detailed guidance on how to develop user's own agent designs and tasks based on AgentSquare, demonstrating the high extensibility of AgentSquare framework.

I believe AgentSquare's novel framework and good usability make it a valuable addition to your repository. Thank you for considering this request.
